### PR TITLE
1854 - Added Live validation for input fields

### PR DESCRIPTION
**Pull request by teodormoq**

### CODE REVIEW

File: src/app/component/Field/Field.component.js [hunk 1]
```diff
@@ -240,13 +241,14 @@ export class Field extends PureComponent {
...
-            message
+            message,
+            validationStatus
         } = this.props;
 
         return (
             <div
               block="Field"
-              mods={ { type, hasError: !!message } }
+              mods={ { type, hasError: validationStatus === false || !!message, isValid: validationStatus === true } }
```
Comment (yeegor) [inline src/app/component/Field/Field.component.js:251]: Please make this multiline, it is very long

File: src/app/style/base/_input.scss [hunk 2]
```diff
@@ -36,6 +37,11 @@ select {
             margin-top: 1.4rem;
         }
     }
+
+    &:-webkit-autofill {
+        box-shadow: 0 0 0 30px var(--input-background-color) inset !important;
+        background-color: var(--input-background-color) !important;
+    }
```
Comment (yeegor) [inline src/app/style/base/_input.scss:44]: The project uses `autoprefixer`, I believe you have no need to prefix for specific browsers separately.

File: src/app/component/Field/Field.container.js [hunk 3]
```diff
@@ -129,20 +138,65 @@ export class FieldContainer extends PureComponent {
...
+            validationStatus
         };
     };
 
+    validateField() {
+        const {
+            validation, id, formRef: refMap
+        } = this.props;
+
+        if (validation && id && refMap && refMap.current) {
```
Comment (yeegor) [inline src/app/component/Field/Field.container.js:157]: It's better to inverse this if.
```js
if (!validation || !id || !refMap || !refMap.current) {
  return;
}
... // the logic from the original if statement
```

File: src/app/component/Field/Field.container.js [hunk 4]
```diff
@@ -129,20 +138,65 @@ export class FieldContainer extends PureComponent {
...
+                    return false;
+                }
+                const validationRules = validationConfig[rule];
+                const isValid = validationRules.validate(inputNode, refMap);
+                return !isValid;
+            });
+
+            if (rule) {
+                return validationConfig[rule];
+            }
```
Comment (yeegor) [inline src/app/component/Field/Field.container.js:171]: Please provide comments on this, this is not an obvious piece of logic.

**Comment by alfredsgenkins** [comment]
Migrated manually! Thanks for your contribution!

**Pull request closed without merging**